packages/teleport-plugin-next-static-props/src/index.ts

```diff
@@ -19,25 +18,49 @@ export const createStaticPropsPlugin: ComponentPluginFactory<StaticPropsPluginCo
   const { componentChunkName = 'jsx-component' } = config || {}
 
   const staticPropsPlugin: ComponentPlugin = async (structure) => {
-    const { uidl, chunks, options } = structure
+    const { uidl, chunks, options, dependencies } = structure
     const { resources } = options
 
-    if (!uidl.outputOptions?.initialPropsData || !resources?.items) {
+    if (!uidl.outputOptions?.initialPropsData) {
       return structure
     }
 
     const { resource } = uidl?.outputOptions?.initialPropsData
-    const usedResource = resources.items[resource.id]
 
-    if (!usedResource) {
-      throw new TeleportError(
-        `Resource ${resource.id} is being used, but missing from the project ressources. Check ${uidl.name} in UIDL for more information`
+    const isLocalResource = 'id' in resource
+    const isExternalResource = 'name' in resource
+    /*
+      Name of the function that is being imported
+    */
+    let resourceImportName
+
+    if (isLocalResource) {
+      const usedResource = resources.items?.[resource.id]
+      if (!usedResource) {
+        throw new TeleportError(
+          `Resource ${resource.id} is being used, but missing from the project ressources. Check ${uidl.name} in UIDL for more information`
+        )
+      }
+      resourceImportName = StringUtils.dashCaseToCamelCase(
+        StringUtils.camelCaseToDashCase(`${usedResource.name}-resource`)
       )
+
+      const importPath = `${resources.path}${StringUtils.camelCaseToDashCase(usedResource.name)}`
+
+      dependencies[resourceImportName] = {
+        path: importPath,
+        /*
+          Local resources can be marked as a library. As they will be ignored in adding to package.json at the end.
+        */
+        type: 'library',
+        version: 'latest',
```

Comment:
I just realised, we can mark the type as `local` for local dependency. So, we won't accidentally mark local imports as deps.

```js
{
  path: '../.../<function>',
  type: 'loca'
}
```

And for external like caisey or others we can mark as

```js
{
  type: 'package',
  version: '..',
  path: '<absolute-path>'
}
```

Reply:
You mean to replace library with local?